src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy

```diff
@@ -88,8 +88,9 @@ class JcrNodesWriter implements ItemWriter<ProtoNode>, ItemWriteListener {
     }
 
     private static void writeToJcr(ProtoNode nodeProto, Session session) {
-        JcrNodeDecorator jcrNode = new ProtoNodeDecorator(nodeProto).writeToJcr(session)
-        jcrNode.setLastModified()
+        ProtoNodeDecorator protoNodeDecorator = new ProtoNodeDecorator(nodeProto)
+        JcrNodeDecorator nodeDecorator = new JcrNodeDecorator(protoNodeDecorator.getOrCreateNode(session))
+        nodeDecorator.writeToJcr(protoNodeDecorator, session)
         // This will processed all mandatory child nodes only
         if(nodeProto.mandatoryChildNodeList && nodeProto.mandatoryChildNodeList.size() > 0) {
```

Comment:
_"From what I thought was that all the JCR operations should be under the Jcr decorator classes and Proto Decorators should be used for proto related operations only"_ 

This will probably be mostly true. What I'm referencing above doesn't conform to this. We have a third actor creating decorators; passing decorators to decorators; having some decorators write their own nodes; having third actor inspecting decorators, and writing nodes elsewhere...

The point is to have this logic in a place where it is cohesive. In other words, it tends to change with the things that change around it.